stdlib/public/core/LifetimeManager.swift

```diff
@@ -283,3 +283,40 @@ extension String {
 public func _copy<T>(_ value: T) -> T {
   copy value
 }
+
+/// Unsafely discard any lifetime dependency on the `dependent` argument. Return
+/// a value identical to `dependent` with a lifetime dependency on the caller's
+/// borrow scope of the `borrows` argument.
+///
+/// TODO: Remove @_unsafeNonescapableResult. Instead, the unsafe dependence
+/// should be expressed by a builtin that is hidden within the function body.
+@_unsafeNonescapableResult
+@_alwaysEmitIntoClient
+@inline(__always)
+@lifetime(borrow source)
+internal func unsafeLifetime<T: ~Copyable & ~Escapable,
```

Comment:
```suggestion
internal func _unsafeLifetime<T: ~Copyable & ~Escapable,
```
internal symbols should have a leading underscore.

Reply:
I see that's commonly done. Was hoping to find out through review whether it's purely stylistic or whether there's an important reason for the underscore.

Reply:
We call this the [Leading Underscore Rule](https://github.com/swiftlang/swift/blob/main/docs/StandardLibraryProgrammersManual.md#the-leading-underscore-rule), and it is a fundamental tool in managing the stdlib's public API surface. We strictly reserve non-underscored names for Swift Evolution approved public API; there are no exceptions, no ifs, no buts.

> This rule ensures we don't accidentally clutter the public namespace with @usableFromInline things (which could prevent us from choosing the best names for newly public API later), and it also makes it easy to see at a glance if a piece of stdlib code uses any non-public entities.

Note that things declared purely internal/private or `@_alwaysEmitIntoClient` have a remarkable tendency to end up shipping as `@usableFromInline internal` or even `public`. Our insistence on applying the Leading Underscore Rule _everywhere_ protects us from massive pain, as it forces every adopter to use underscored names right from the start.

As a general rule, no API survives the Swift Evolution process unchanged. By forcing early clients to consistently use underscored names, we allow the S-E process to experiment with arbitrarily renaming/rearranging things without forcing all those prototyping use cases to immediately go through a confusing migration process after every change.

Reply:
Oh, I like the rule. Just wanted to test it because I recently heard that feature guards made the rule obsolete. But we still have
- guard against accidental visibility
- stylistically very helpful